lib/Model/Dao/PimcoreLocationAwareConfigDao.php

```diff
@@ -63,13 +65,17 @@ protected function getDataByName(string $id)
     {
         $this->id = $id;
 
-        if (isset(self::$cache[$this->settingsStoreScope][$id])) {
+        if (isset(self::$cache[$this->settingsStoreScope][$id]) &&
+            isset(self::$cache[$this->settingsStoreScope][$id][self::cacheKeyDataSource])) {
+
+            $this->dataSource = self::$cache[$this->settingsStoreScope][$id][self::cacheKeyDataSource];
             return self::$cache[$this->settingsStoreScope][$id];
         }
 
         list($data, $this->dataSource) = $this->locationAwareConfigRepository->loadConfigByKey($id);
 
         self::$cache[$this->settingsStoreScope][$id] = $data;
+        self::$cache[$this->settingsStoreScope][$id][self::cacheKeyDataSource] = $this->dataSource;
```

Comment:
```suggestion
        self::$cache[$this->settingsStoreScope][$id] = [
            'datasource' => $this->dataSource,
            'data' => $data,
        ];
```

This assumes `$data` is an array, it would also override a config node called `datasource`. so I'd wrap this in another array container. I think we can then as well live without having `datasource` as a constant 😊